lls/main.c

```diff
@@ -893,9 +893,18 @@ lls_stage2(void *arg)
 		}
 
 		/* Receive ICMP packets using IPv4 ACL filters. */
-		ret = register_icmp_acl_rules(&net_conf->front);
-		if (ret < 0)
-			return ret;
+		if (ipv4_if_configured(&net_conf->front)) {
```

Comment:
If ARP is enabled, why do we still need to test for IPv4?

This comment applies to the other tests in this patch as well as for ND for IPv6 cases.

Reply:
Sorry, I thought there was a bug here. I'm going to drop this commit.